tests/test_name_server.rb

```diff
@@ -0,0 +1,120 @@
+#
+# Minitest for NameServer class
+#
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require File.expand_path('../ciscotest', __FILE__)
+require File.expand_path('../../lib/cisco_node_utils/name_server', __FILE__)
+
+# TestNameServer - Minitest for NameServer node utility.
+class TestNameServer < CiscoTestCase
+  def setup
+    # setup runs at the beginning of each test
+    super
+    no_nameserver_google
+  end
+
+  def teardown
+    # teardown runs at the end of each test
+    no_nameserver_google
+    super
+  end
+
+  def no_nameserver_google
+    # Turn the feature off for a clean test.
+    @device.cmd('conf t ; no ip name-server 8.8.8.8 ; end')
+    @device.cmd('conf t ; no ip name-server 2001:4860:4860::8888 ; end')
+    # Flush the cache since we've modified the device outside of the node_utils APIs
+    node.cache_flush
+  end
+
+  # TESTS
+
+  def test_nameserver_create_destroy_single_ipv4
+    assert_equal({}, Cisco::NameServer.nameservers,
```

Comment:
That'd be the most thorough approach (we have something similar in `test_snmp_user.rb` for a similar reason). Alternately as a simpler approach, you could not worry about whatever is pre-existing and only check the specific instances you're testing - something like:

```
refute_includes(Cisco::NameServer.nameservers, id)
...
# create instance ...
...
assert_includes(Cisco::NameServer.nameservers, id)
assert_equal(Cisco::NameServer.nameservers[id], ns)
...
# destroy instance
...
refute_includes(Cisco::NameServer.nameservers, id)
```